java/piranha/src/main/java/com/uber/piranha/XPFlagCleaner.java

```diff
@@ -139,12 +153,12 @@
    */
   private int endPos = DONTCARE;
 
-  /* information provided in the config file */
-  private final HashSet<String> treatedMethods = new HashSet<String>();
-  private final HashSet<String> controlMethods = new HashSet<String>();
-  private final HashSet<String> deleteMethods = new HashSet<String>();
-  private final HashSet<String> treatmentGroupMethods = new HashSet<String>();
-  private final HashSet<String> handledAnnotations = new HashSet<String>();
+  /** Information provided in the properties.json config file */
+  private final Set<Map<String, Object>> methodProperties = new HashSet<>();
+
+  private final Map<String, Map<String, List<Map<String, Object>>>> configMethodsMap =
+      new HashMap<>();
```

Comment:
I think this can be a map from methodname to a set of attributes. Something like: 

methodX -> {(treated, bool, com.uber.*)}
methodY -> {(control, bool, com.uber.*), (control, bool, some_other_receiver), ...}

The idea is to do one hashmap lookup instead of four different lookups for each method call. In most cases, there won't be a match and so this will be faster.

Reply:
This is not currently true. The first key to `configMethodsMap` is the flag type. See [these calls](https://github.com/uber/piranha/pull/39/files#diff-ddde662a04227b8ea9bef9fc02110791R327) and the target method implementation.

If we do go with the approach you are proposing (which makes sense), then: a) the top level key would need to be the method, b) I'd change `Map<String, List<Map<String, Object>>>` to `ImmutableList<PiranhaMethodRecord>` or something like that, where `PiranhaMethodRecord` is some sort of typed config record with fields like `.flagType`, `.receiverType`, and so on.

Btw, extra pedantic nit: I'd prefer `ImmutableMap` and `ImmutableList` here and builders. But since the original code was also using mutable collections, I am absolutely fine leaving that for a follow up issue you two can assign to me 😅

Reply:
+1.